src/services/riotApiClient.js

```diff
@@ -109,27 +114,68 @@ class RiotApiClient {
       return { 
         matchDate: match.info.game_datetime,
         matchLength: match.info.game_length,
-        participants: match.metadata.participants
+        matchId: match.metadata.match_id
       }
     })
   
-    matches.forEach (async match => {
-      match.xparticipants = await Promise.all(match.participants.map(participant => this.fetchTFTSummonerInfoByPuuid(participant)))
-    })   
     return matches
     
   }
+
+  async transformMatchData(matchInfo) {
+    const summoners = await Promise.all(matchInfo.info.participants.puuid.map( puuid => this.fetchTFTSummonerInfoByPuuid(puuid)))
+      return {
+      matchId: matchInfo.metadata.match_id,
+      matchTime: matchInfo.info.game_datetime,
+      matchLength: matchInfo.info.game_length,
+      matchParticipants: summoners
+      }
+    return newMatchInfo
+  }
   
   transformSummonerInfo(summonerInfo) {
     const iconID = summonerInfo.profileIconId
-    const url = `http://ddragon.leagueoflegends.com/cdn/${this.latestLolPatchVersion}/img/profileicon/${iconID}.png`;
+    const url = `http://ddragon.leagueoflegends.com/cdn/${this.latestLolPatchVersion}/img/profileicon/${iconID}.png`
     return {
-      name: summonerInfo.name,
+      summonerName: summonerInfo.name,
       level: summonerInfo.summonerLevel, 
-      profileIconUrl: summonerInfo.profileIconId,
+      profileIconUrl: url,
       puuid: summonerInfo.puuid
     }
   }
+  
+  async correctMatchTransformation(matchInfo) {
+    const relevantMatchInfo = await Promise.all(matchInfo.info.participants.map( async info => {
```

Comment:
Since you're just returning it on the first line after this statement, get rid of the const and just return this.
```suggestion
    return await Promise.all(matchInfo.info.participants.map( async info => {
```